File: app/src/main/java/com/mss/instamat/repositories/network/NetMapper.java
package com.mss.instamat.repositories.network;

import android.support.annotation.NonNull;

import com.mss.instamat.domain.models.ImageInfo;
import com.mss.instamat.repositories.network.models.ImageInfoNet;

import java.util.List;
import java.util.stream.Collectors;

class NetMapper {

    @NonNull
    public static List<ImageInfo> mapFromNet(@NonNull final List<ImageInfoNet> imagesNet) {
        return imagesNet
                .stream()
                .map(imageInfoDB -> mapFromNet(imageInfoDB))
                .collect(Collectors.toList());
    }

    @NonNull
    private static ImageInfo mapFromNet(@NonNull final ImageInfoNet imageInfoDB) {
        final ImageInfo imageInfo = new ImageInfo();
        imageInfo.setId(imageInfoDB.getId());
        imageInfo.setLargeImageURL(imageInfoDB.getLargeImageURL());
        imageInfo.setPreviewURL(imageInfoDB.getPreviewURL());
        return imageInfo;
    }
}

File: app/src/main/java/com/mss/imagesearcher/model/repositories/network/NetMapper.kt
package com.mss.imagesearcher.model.repositories.network

import com.mss.imagesearcher.model.entity.ImageInfo
import com.mss.imagesearcher.model.repositories.network.entity.ImageInfoNet
import java.util.*

object NetMapper {

    fun mapFromNet(imagesNet: List<ImageInfoNet>): List<ImageInfo> {
        val images = ArrayList<ImageInfo>()
        for (imageNet in imagesNet) {
            images.add(mapFromNet(imageNet))
        }
        return images
    }

    private fun mapFromNet(imageInfoDB: ImageInfoNet) = ImageInfo(
            imageInfoDB.id,
            imageInfoDB.largeImageURL,
            imageInfoDB.previewURL,
            imageInfoDB.webFormatURL,
            imageInfoDB.height,
            imageInfoDB.width,
            imageInfoDB.type,
            imageInfoDB.tags,
            imageInfoDB.pageUrl,
            imageInfoDB.likes,
            imageInfoDB.views,
            imageInfoDB.comments
    )
}